src/keyring.ts

```diff
@@ -1,7 +1,14 @@
+import type { Transaction, TxData } from '@ethereumjs/tx';
+import type {
+  MessageTypes,
```

Comment:
Thoughts on moving these types into this library? If we import them, we'd be creating a circular dependency

Reply:
We should probably try to sort it out in the short term to avoid any future headaches. One option could be to unify both modules into `@metamask/utils`, I don't see any strong reasons to have two `utils` modules.

Reply:
Sorry I don't follow what you mean here. Do you mean we should just use a circular dependency for now? They are allowed by Yarn/npm but it's pretty common for development/build tooling to run into issues with them.

I don't think it would be appropriate to combine the two packages. `eth-sig-util` isn't just another utility library, it's focused solely on signatures. It's an open source package that has widespread use outside of MetaMask as well. It also has a few "heavier" dependencies that we don't necessarily want to add to every single other package.

Reply:
Yeah, sorry for not being clear. In meant that in my last message, allow the circular dependency and, in the future, combine both packages. But you made good points there so we should avoid going down that path.

I personally would like to avoid the circular dependency problem.